app/models/piece.rb

```diff
@@ -31,7 +31,7 @@ def blocked?(x, y)
     y_direction = y <=> y_position
 
     # how many times will the check occur
-    steps = delta_x > delta_y ? delta_x : delta_y
+    steps = delta_x(x_position, x) > delta_y(y_position, y) ? delta_x(x_position, x) : delta_y(y_position, y)
```

Comment:
One idea to reduce complexity here would be to save the calls to `delta_x` and `delta_y` as variables and then use those saved values, to avoid calling those methods multiple times:

```ruby
dx = delta_x(x_position, x)
dy = delta_y(y_position, y)
steps = dx > dy ? dx : dy
```

Ideally that should fix it, otherwise we could increase the complexity tolerance in `.rubocop.yml` from 20 to something like 25.